lib/daemon/index.js

```diff
@@ -1,22 +1,56 @@
 // Command manager for daemonization
 
+var chalk = require('chalk');
+var cluster = require('cluster');
 var fs = require('fs');
+var net = require('net');
 var path = require('path');
-var cluster = require('cluster');
+var rimraf = require('rimraf');
+var xpipe = require('xpipe');
+
 var interrupt = require('./interrupt');
-var chalk = require('chalk');
 var mage = require('lib/mage');
 
 
 // Daemon configuration
 
 mage.core.config.setTopLevelDefault('daemon', path.join(__dirname, 'config.yaml'));
 
-var APP_LOCK_FILE = mage.core.config.get(['daemon', 'pidfile']);
-var CMD_LOCK_FILE = './.commandpidfile';
+const APP_LOCK_FILE = mage.core.config.get(['daemon', 'pidfile']);
+const CMD_LOCK_FILE = './.commandpidfile';
+const STDIO_PIPE_PATH = xpipe.eq('./.commandstdio');
 
 var timeouts = mage.core.config.get(['daemon', 'timeouts']) || {};
 
+function disableDaemonizedStdio(commandPid) {
+	if (!commandPid) {
+		return;
+	}
+
+	// Send stdio to a black hole
+	function disable(io) {
+		io.write = () => true;
+	}
+
+	disable(process.stdout);
+	disable(process.stderr);
+}
+
+function setDaemonizedStdio(commandPid) {
+	if (!commandPid) {
```

Comment:
The PID is not relevant to this function, so to pass it... is quite confusing (I know, because I've been trying to make sense of this PR in review ;))

Would it not make more sense to have a boolean returning function somewhere (instead of a PID, but by all means on the same basis), and if that boolean is false, to not *call* functions like (un)setDaemonizedStdio?

Reply:
The issue is that you then end up having to carry a conditional *everywhere* (well. 4 places). 

I'll try to make this look a bit clearer.